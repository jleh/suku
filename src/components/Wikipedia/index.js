import React from 'react';

export default ({ person }) => person.wikipedia && (
  <div>
    <a href={person.wikipedia} target="_blank" rel="noopener noreferrer">
      Wikipedia
    </a>
  </div>
);
