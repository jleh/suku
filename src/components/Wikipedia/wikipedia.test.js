import React from 'react';
import renderer from 'react-test-renderer';

import Wikipedia from '.';

const renderComponent = (person) => renderer.create(<Wikipedia person={person} />).toJSON();

test('Renders wikipedia link if person has wikipedia page', () => {
  const person = { wikipedia: 'http://fi.wikipedia.fi' };
  expect(renderComponent(person)).toMatchSnapshot();
});

test('Empty if person has no wikipedia link', () => {
  const person = {};
  expect(renderComponent(person)).toMatchSnapshot();
});
