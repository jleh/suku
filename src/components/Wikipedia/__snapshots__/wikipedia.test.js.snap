// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty if person has no wikipedia link 1`] = `null`;

exports[`Renders wikipedia link if person has wikipedia page 1`] = `
<div>
  <a
    href="http://fi.wikipedia.fi"
    rel="noopener noreferrer"
    target="_blank"
  >
    Wikipedia
  </a>
</div>
`;
